apps/web/src/components/search/search.tsx

```diff
@@ -0,0 +1,186 @@
+'use client';
+
+import * as React from 'react';
+
+import { SearchIcon, Loader2 as LoaderIcon } from '@repo/ui/icons';
+import { Button } from '@repo/ui/components/button';
+import { Separator } from '@repo/ui/components/separator';
+import { Text } from '@repo/ui/components/typography/typography';
+import { ScrollArea, ScrollBar } from '@repo/ui/components/scroll-area';
+import { Dialog, DialogContent, DialogTrigger } from '@repo/ui/components/dialog';
+import Link from 'next/link';
+import { DifficultyBadge } from '@repo/ui/components/difficulty-badge';
+import { Markdown } from '@repo/ui/components/markdown';
+import type { Challenge } from '@repo/db/types';
+import { SearchBoxContextProvider, useSearchBox } from './searchbox.context';
+import { SearchProvider, useSearchResult, useSearchStatus } from './search-provider';
+
+export function Search() {
+  const [open, setOpen] = React.useState(false);
+
+  React.useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
```

Comment:
Maybe we should modify this slightly so that the user can press this shortcut again to close the modal.

Reply:
Look at the useEffect in the Readme.md here: https://github.com/pacocoursey/cmdk

Reply:
@bautistaaa  recent searches implemented